### ISSUE TITLE
Steam says there are SteamApps and steamapps folders, and asks me to manually fix it by deleting SteamApps, but it isn't there appearently

### ISSUE THREAD

**Issue by ugurcansayan** [ValveSoftware/steam-for-linux#7665]
Sorry for my bed England! JK but English isn't my mother tongue, so please pardon me.

#### Your system information

* Steam client version (build number or date): Feb 12 2021, 23:22:26
* Distribution (e.g. Ubuntu): Linux Mint 20.1 x86_64 
* Opted into Steam client beta: No
* Have you checked for system updates: Yes
* Happens with both kernels: 5.4.0-65 and 5.8.0-43

#### Please describe your issue in as much detail as possible:

Steam for Linux (installed from mintinstall https://community.linuxmint.com/software/view/steam:i386 ). 
Formatted an HDD as exFAT and connected it via USB 3.0. Created a folder in it only for SteamLibrary. On Steam client startup, it says (can be selected, copied)
```
/media/ugurcan/Ugurcan/Oyun/SteamLibrary has both 'SteamApps' and 'steamapps' directories.

This will cause problems. Please fix manually and only keep 'steamapps'
``` 
There is no such folder called SteamApps. 

![Ekran Görüntüsü - 2021-02-20 00-46-25](https://user-images.githubusercontent.com/25729919/108565166-4ab7d080-7315-11eb-948e-ad9ee46a176d.png)
_Hidden files are set to be shown._ 

I tried renaming folders, searching for SteamApps on my system, creating a new folder in same drive, but none of them worked. Also, using client after this warning pop-up, downloading games and launching games are OK. When I plug same HDD on Windows and launch Steam client, it doesn't say anything, it just works there too.

#### Steps for reproducing this issue:

1. Format an HDD ext4
2. Open Steam client settings, choose download location and create folder in it
3. Click OK.
4. Quit Steam.
5. Launch Steam client.
6. Warning pops up.
7. Repeat 4.

**Comment by kisak-valve** [comment]
Adding a couple keywords, please ignore this message: Camel-case, lower-case

**Comment by TTimo** [comment]
This is likely caused by exFAT and would happen with other case insensitive filesystems. The Steam client currently assumes it is on a case sensitive filesystem.

**Comment by ugurcansayan** [comment]
> This is likely caused by exFAT and would happen with other case insensitive filesystems. The Steam client currently assumes it is on a case sensitive filesystem.

Hello, thanks for reply, I didn't know that fact. For some reason I assumed exFAT handled case-sensivity the same way as NTFS.

Also I noticed another bug (probably) relating to exFAT and Steam-Linux, should I open a new issue or edit/update this one?

**Comment by ctubbsii** [comment]
I'm using exFAT also and getting the same issue. It's very annoying, because I know there's only one folder, but it seems harmless.

**Comment by UbioZur** [comment]
Same here, on exFat (so easy sharing of the library install between win10 and Debian).
Same message at Steam startup when there is only one folder "steamapps" (which comes from Win10)

**Comment by neffisaurus** [comment]
I have the same issue on an ext4 filesystem! I do have some Windows games installed via proton. The maddening thing is that the steam client keeps installing games into SteamApps rather than steamapps anyway. All my games, proton and native, appear to work fine, so I'm guessing it's just a misplaced error?

**Comment by TTimo** [comment]
@neffisaurus this would be a different problem, please make a separate report.

**Comment by aherstein** [comment]
Same issue here, build date 2021-05-19, 5.11.0-7614-generic kernel
My secondary library mounted locally from a NAS using cifs and it's for that library that I get the error message.

**Comment by TheJuicyOne** [comment]
I have the exact same problem

**Comment by G4biel** [comment]
Oi, pessoa! Estou com o mesmo problema, o que devo fazer para resolvê-lo?

![Screenshot 2021-07-12 12 52 55](https://user-images.githubusercontent.com/87324587/125318576-5762b280-e310-11eb-9e80-e8518bc9840b.png)

![Screenshot 2021-07-12 12 53 03](https://user-images.githubusercontent.com/87324587/125318669-6f3a3680-e310-11eb-8010-d610029ad007.png)

![Screenshot 2021-07-12 12 53 29](https://user-images.githubusercontent.com/87324587/125318687-75301780-e310-11eb-913e-394c4c246136.png)

**Comment by TheJuicyOne** [comment]
yes im on chromebook too, i  guess we have the exact same problem

**Comment by G4biel** [comment]
Encontrou alguma solução?

**Comment by TheJuicyOne** [comment]
i couldnt find one we may just have to wait for it to be fixed or something

**Comment by scorpius2k1** [comment]
I have this issue as well persisting for over 6 months now on multiple Linux distributions, including supported Debian/Ubuntu.

**Comment by Crono141** [comment]
I also have this issue.  Same circumstances (exFAT drive/folder).  Seems the check script that steam runs just needs to be updated to either become aware of the FS its on, or check that both folders actually exist (so that a single folder doesn't trigger this issue).

It is a harmless error.  Maybe a tick box to suppress this message in the future, barring an actual fix for the check.

**Comment by oddstr13** [comment]
This issue is also present for case insensitive cifs mounts (which seems to be caused by an old bug in cifs) https://superuser.com/questions/680492/mounts-arent-case-sensitive

The best way for steam to deal with this, would probably be to check the directory listing instead of checking if both paths exist.
(Python example as psudocode)
```python
# Current behavior:
if os.path.exists("/foo/SteamLibrary/SteamApps") and os.path.exists("/foo/SteamLibrary/steamapps"):
    error("Can't have both")

# More reliable behavior (for case insensitive filesystems):
listing = os.listdir("/foo/SteamLibrary")
if "SteamApps" in listing and "steamapps" in listing:
    error("Can't have both")
```

**Comment by scorpius2k1** [comment]
[This is an automated-reply to your message]
     Greetings,
    I am currently away on vacation from August 2–15 and will be "AFK"
    during that time. If your message can wait, I will respond upon my
    return. If this is an urgent issue, please send a message with
    *URGENT* in the subject and I will respond as soon as possible. 
    
    
    Scorp
    
    "This tram ride is sweet. I can't imagine how horrible it would
      be to have to walk all the way out of here." - Gordon Freeman

**Comment by michielbaird** [comment]
Having this issue after mounting a smb share on my steam deck

**Comment by benterova** [comment]
Having this issue on a fresh install of openSUSE. Interestingly, the error only shows on the native installed Steam, while the flatpak version fails silently.

**Comment by sylv256** [comment]
Having this issue when trying to share my Games drive with Windows. I guess I've learned my lesson.
